django/contrib/auth/forms.py

```diff
@@ -177,30 +151,82 @@ def validate_passwords(
 
     def validate_password_for_user(self, user, password_field_name="password2"):
         password = self.cleaned_data.get(password_field_name)
-        if password and self.cleaned_data["set_usable_password"]:
+        if password:
             try:
                 password_validation.validate_password(password, user)
             except ValidationError as error:
                 self.add_error(password_field_name, error)
 
     def set_password_and_save(self, user, password_field_name="password1", commit=True):
+        user.set_password(self.cleaned_data[password_field_name])
+        if commit:
+            user.save()
+        return user
+
+
+class SetUnusablePasswordMixin:
+    """
+    Form mixin that allows setting an unusable password for a user.
+
+    This mixin should be used in combination with `SetPasswordMixin`.
+    """
+
+    usable_password_help_text = _(
+        "Whether the user will be able to authenticate using a password or not. "
+        "If disabled, they may still be able to authenticate using other backends, "
+        "such as Single Sign-On or LDAP."
+    )
+
+    @staticmethod
+    def create_usable_password_field(help_text=usable_password_help_text):
+        return forms.ChoiceField(
+            label=_("Password-based authentication"),
+            required=False,
+            initial="true",
+            choices={"true": _("Enabled"), "false": _("Disabled")},
+            widget=forms.RadioSelect(attrs={"class": "radiolist inline"}),
+            help_text=help_text,
+        )
+
+    def validate_passwords(
+        self,
+        *args,
+        usable_password_field_name="usable_password",
+        **kwargs,
+    ):
+        usable_password = (
+            self.cleaned_data.pop(usable_password_field_name, None) != "false"
+        )
+        self.cleaned_data["set_usable_password"] = usable_password
+
+        if usable_password:
+            super().validate_passwords(*args, **kwargs)
+
+    def validate_password_for_user(self, user, **kwargs):
         if self.cleaned_data["set_usable_password"]:
-            user.set_password(self.cleaned_data[password_field_name])
+            super().validate_password_for_user(user, **kwargs)
+
+    def set_password_and_save(self, user, commit=True, **kwargs):
+        if self.cleaned_data["set_usable_password"]:
+            user = super().set_password_and_save(user, **kwargs, commit=commit)
         else:
             user.set_unusable_password()
-        if commit:
-            user.save()
+            if commit:
+                user.save()
         return user
 
 
 class BaseUserCreationForm(SetPasswordMixin, forms.ModelForm):
     """
     A form that creates a user, with no privileges, from the given username and
     password.
+
+    This is the documented and recommended base class when there is a need to
+    customize the user creation form, so it should be kept simple and unchanged
+    (except bugs of course).
```

Comment:
I think instead of this doc string update, we should add a regression test which checks that fields
```diff
--- a/tests/auth_tests/test_forms.py
+++ b/tests/auth_tests/test_forms.py
@@ -80,6 +80,13 @@ class BaseUserCreationFormTest(TestDataMixin, TestCase):
 
     form_class = BaseUserCreationForm
 
+    def test_form_fields(self):
+        form = self.form_class()
+        self.assertEqual(
+            list(form.fields.keys()),
+            ["username", "password1", "password2"]
+        )
+
     def test_user_already_exists(self):
         data = {
             "username": "testclient",
@@ -1415,6 +1422,13 @@ class AdminUserCreationFormTest(BaseUserCreationFormTest):
 
     form_class = AdminUserCreationForm
 
+    def test_form_fields(self):
+        form = self.form_class()
+        self.assertEqual(
+            list(form.fields.keys()),
+            ["username", "password1", "password2", "usable_password"]
+        )
+
     @override_settings(
```
Then if anyone was to add fields to `BaseUserCreationForm`, `BaseUserCreationFormTest.test_form_fields` would break

Reply:
Great idea, thank you, applying!